### DISCUSSION TITLE
Creating DeveloperMetadata

### DISCUSSION THREAD

**Discussion by darinw** [theoephraim/node-google-spreadsheet#634] [General]
I would like to create, update and retrieve Developer Metadata. I see there are a few functions that are unimplimented.
![image](https://github.com/theoephraim/node-google-spreadsheet/assets/92472/d6925cda-486f-49f2-82dd-681bfdbcead2)

do you have plans to add that functionality? If not is there a recommended way of using this library where i can but use the official library where not? ideally via this library exposing the official api?

**Comment by theoephraim** [comment]
Hi @darinw!

Yes - I'd like to support pretty much everything the native API does eventually. It's just a matter of prioritization and having free time... The more folks ask for something, definitely the more likely it is I'll implement it over other things. Now that I've finally released a major rewrite in typescript, I should have some time to put towards specific features.

The "official API" is just making api requests according to the [official docs](https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets/request#createdevelopermetadatarequest). I've just tried to make the functions a bit more user friendly wherever possible...

If you dig through the code for [GoogleSpreadsheet](https://github.com/theoephraim/node-google-spreadsheet/blob/master/src/lib/GoogleSpreadsheet.ts) you can see how other similar requests are being made.

Probably the best thing to do would be implement the methods and open a PR.

I will eventually tackle it and implement myself, but just can't make any promises on timing.
